### PR TITLE
TMEDIA-400 subs POC

### PR DESCRIPTION
**Pull request by mattkim93**
**If you have not filled out the checklist below, the pr is not ready for review.**

## Description
_Information about what you changed for this PR_
- Adds Identity blocks
- Adds dummy paywall feature in Subscriptions blocks

## Jira Ticket
- [TMEDIA-400](https://arcpublishing.atlassian.net/browse/TMEDIA-400)

## Test Steps
- Add test steps a reviewer must complete to test this PR

See above paper doc for test steps

### Before
_Example: When I clicked the search button, the button turned red._

[include screenshot or gif or link to video, storybook would be awesome]

### After
_Example: When I clicked the search button, the button turned green._

[include screenshot or gif or link to video, storybook would be awesome]

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- See above paper doc

## Author Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
- [ ] Confirmed all the test steps a reviewer will follow above are working.
- [ ] Confirmed there are no linter errors. Please run `npm run lint` to check for errors. Often, `npm run lint:fix` will fix those errors and warnings.
- [ ] Ran this code locally and checked that there are not any unintended side effects. For example, that a CSS selector is scoped only to a particular block.
- [ ] Confirmed this PR has reasonable code coverage. You can run `npm run test:coverage` to see your progress.
  - [ ] Confirmed this PR has unit test files
  - [ ] Ran `npm run test`, made sure all tests are passing
  - [ ] If the amount of work to write unit tests for this change are excessive,
please explain why (so that we can fix it whenever it gets refactored).
- [ ] Confirmed relevant documentation has been updated/added.

## Reviewer Checklist
_The reviewer of the PR should copy-paste this template into the review comments on review._

- [ ] Linting code actions have passed.
- [ ] Ran the code locally based on the test instructions.
  - [ ] I don’t think this is needed to be tested locally. For example, a padding style change (storybook?) or a logic change (write a test).
- [ ] I am a member of the engine theme team so that I can approve and merge this. If you're not on the team, you won't have access to approve and merge this pr.
- [ ] Looked to see that the new or changed code has code coverage, specifically. We want the global code coverage to keep on going up with targeted testing.

### CODE REVIEW

File: blocks/identity-block/components/Identity.jsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+const useIdentity = () => {
+  // console.log('running hook...');
+  const { arcSite } = useFusionContext();
+  const { subscriptions } = getProperties(arcSite);
+  const [isInit, setIsInit] = useState(() => !!Identity.apiOrigin);
+  // console.log('props...', arcSite, subscriptions, isInit);
+  if (!isInit && arcSite && subscriptions?.identity?.apiOrigin) {
+    const arcHeaders = { 'Arc-Organization': 'staging', 'Arc-Site': 'staging' };
+    if (!isServerSide()) {
+      if (!window.realFetch) {
```
Comment (mattkim93) [inline blocks/identity-block/components/Identity.jsx:16]: For local dev only

File: blocks/subscriptions-block/features/paywall/default.jsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+import { useIdentity } from '@wpmedia/identity-block';
+import './styles.scss';
+
+const Paywall = ({ customFields }) => {
+  const { signUpURL, loginURL } = customFields;
+  const { Identity } = useIdentity();
+  const [showPaywall, setShowPaywall] = useState(false);
+  const [loggedIn, setIsLoggedIn] = useState(false);
+
+  useEffect(() => {
```
Comment (matthewroach) [inline blocks/subscriptions-block/features/paywall/default.jsx:12]: The `useEffect`s in here are for demo purpose. Paywall is another concept item we is too be looked at

File: blocks/header-nav-chain-block/chains/header-nav-chain-block/default.jsx [hunk 3]
```diff
@@ -375,6 +375,8 @@ const Nav = (props) => {
     );
   };
 
+  const rightNavSection = useMemo(() => <NavSection side="right" />, []);
```
Comment (JackHowa) [inline blocks/header-nav-chain-block/chains/header-nav-chain-block/default.jsx:378]: not sure I follow what's happening here. why is `useMemo` necessary here?

File: blocks/identity-block/features/facebook-login/default.jsx [hunk 4]
```diff
@@ -0,0 +1,82 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from '@arc-fusion/prop-types';
+import { isServerSide } from '@wpmedia/engine-theme-sdk';
+
+// eslint-disable-next-line import/extensions
+import useIdentity from '../../components/Identity.jsx';
```
Comment (JackHowa) [inline blocks/identity-block/features/facebook-login/default.jsx:6]: think the `.jsx` can be removed here 

```suggestion
import useIdentity from '../../components/Identity';
```

File: blocks/identity-block/features/facebook-login/default.jsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  }, []);
+
+  return (
+    <main>
+      <section>
+        {isInitialized ? (
+          <>
+            <div
+              className="fb-login-button"
+              data-width="512"
```
Comment (JackHowa) [inline blocks/identity-block/features/facebook-login/default.jsx:46]: is there documentation for some of these fields? I don't know the fb api

File: blocks/identity-block/features/forgot-password/request.jsx [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+            id="username"
+            type="text"
+            onChange={(e) => setUsername(e.target.value)}
+          />
+        </div>
+        <button type="submit">Request reset link</button>
+        {error ? (
+          <section>
+            <p>{`Error requesting password reset. Code: ${error.code}`}</p>
+            {error.message && error.message.length < 100 && <p>{error.message}</p>}
```
Comment (JackHowa) [inline blocks/identity-block/features/forgot-password/request.jsx:42]: shorter than 100 char error message? is there something we need to worry about for showing this to customers?

File: blocks/identity-block/features/header-account-action/default.jsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+          setUser(userProfile);
+        }
+      }).catch((e) => {
+        if (isActive) {
+          setError(e);
+        }
+      });
+    }
+
+    // cancel subscription to useEffect
```
Comment (JackHowa) [inline blocks/identity-block/features/header-account-action/default.jsx:40]: nice

**Review by JackHowa — COMMENTED** [review]
Update: ah I found the test steps https://paper.dropbox.com/doc/Themes-Subscriptions-Proof-of-Concept--BP6mtohMlZkCyVwpnR~P007zAg-aHjkUr5TYxztGDAPBf4eb#:uid=251498380833411576343385&h2=Local-Dev -- didn't see the link

Great start in a short period of time! Have a couple of questions. Is there a good way of getting this up and running for me? I'm curious about local dev instructions

**Review by JackHowa — CHANGES_REQUESTED** [review]
Not really able to replicate the demo based on the dropbox doc. I don't see any login or functionality in the headers. 

I've imported the pages and the templates. I'm on the zscaler vpn. Let me know if you're available to pair @mattkim93

File: blocks/identity-block/components/Identity.jsx [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+import getProperties from 'fusion:properties';
+import { isServerSide } from '@wpmedia/engine-theme-sdk';
+
+const useIdentity = () => {
+  // console.log('running hook...');
+  const { arcSite } = useFusionContext();
+  const { subscriptions } = getProperties(arcSite);
+  const [isInit, setIsInit] = useState(() => !!Identity.apiOrigin);
+  // console.log('props...', arcSite, subscriptions, isInit);
+  if (!isInit && arcSite && subscriptions?.identity?.apiOrigin) {
```
Comment (mattkim93) [inline blocks/identity-block/components/Identity.jsx:13]: from here to line 35 should be taken out

**Comment by JackHowa** [comment]
accepted as done

**Pull request closed without merging**